src/Accounts/Accounts/Rest/InvokeAzRestMethodCommand.cs

```diff
@@ -111,91 +125,309 @@ public override void ExecuteCmdlet()
-                    && !HasSameEndpoint(Uri.Authority, context.Environment.ResourceManagerUrl))
+                // Delay between polling requests; default to 30 seconds if not specified
+                int delay = response.Response.Headers.Contains("Retry-After")
+                    ? int.Parse(response.Response.Headers.GetValues("Retry-After").FirstOrDefault())
+                    : 30;
+
+                Thread.Sleep(TimeSpan.FromSeconds(delay));
+
+                // Polling logic; uses GET method to check operation status
+                string pollingUri = DeterminePollingUri(response);
+                response = serviceClient.Operations.GetResourceWithFullResponse(pollingUri, this.ApiVersion);
+                //WriteObject("New Response Received:" + ": " + response.Response.StatusCode);
+
+                WriteObject(new PSHttpResponse(response));
+
+                if (response.Response.StatusCode == System.Net.HttpStatusCode.Created ||
+                    response.Response.StatusCode == System.Net.HttpStatusCode.Accepted)
                 {
-                    throw new AzPSArgumentException("The resource ID of Azure Resource Manager cannot be used for other endpoint. Please make sure to input the correct resource ID that matches the request URI.",
-                        nameof(ResourceId));
+                    continue;
                 }
-                var targetResourceId = string.IsNullOrEmpty(resourceId) ? MatchResourceId(context, Uri.Authority, out targetResourceIdKey) : resourceId;
-                if (string.IsNullOrWhiteSpace(targetResourceId))
+                else
+                if (response.Response.StatusCode == System.Net.HttpStatusCode.OK)
                 {
-                    throw new AzPSArgumentException("Cannot find resource id(audience) for authentication", nameof(ResourceId));
+                    var error = false;
+                    try
+                    {
+                        string state = GetProvisioningState(response);
+                        if (state is null)
+                        {
+                            // There is insufficient information to continue LRO
+                            // We will return the final request-response
+                            break;
+                        }
+
+                        switch (state?.ToLower())
+                        {
+                            case "failed":
+                                error = true;
+                                break;
+                            case "succeeded":
+                            case "canceled":
+                                break;
+
+                            default:
+                                // Continue Polling
+                                response.Response.StatusCode = System.Net.HttpStatusCode.Created;
+                                continue;
+                        }
+
+                    }
+                    catch {} // Cannot Peek into response
```

Comment:
maybe we need to write warning for the expection.

Reply:
No, we can simply ignore this catch if there is an error since we can fallback to original cmdlet behaviour incase response json is incorrectly formed.